opshin/optimize/optimize_union_expansion.py

```diff
@@ -0,0 +1,220 @@
+from _ast import BoolOp, Call, FunctionDef, If, UnaryOp
+from ast import *
+from typing import Any, List
+from ..util import CompilingNodeTransformer
+from copy import deepcopy
+
+"""
+Expand union types
+"""
+
+
+def type_to_suffix(typ: expr) -> str:
+    try:
+        raw = unparse(typ)
+    except Exception:
+        return "UnknownType"
+    return (
+        raw.replace(" ", "")
+        .replace("[", "_l_")
+        .replace("]", "_r_")
+        .replace(",", "_c_")
+        .replace(".", "_d_")
+    )
+
+
+class RemoveDeadCode(CompilingNodeTransformer):
```

Comment:
```suggestion
class ExpandUnionDefs(CompilingNodeTransformer):
```

Reply:
Or whichever step this is supposed to be? If it is dead code removal why is it not covered by the normal dead code removal yet?

Reply:
If I remember correctly, whatever existing dead code analysis there is wasn't removing redundant isinstance statements. At the very least, the script size and budgets were not coming out as low as I was expecting. I was probably hesitant to modify any existing optimisations to keep this pull request within the scope of the bug bounty and avoid inadvertently introducing bugs or side affects.  I might have missed something simple that means this code is redundant.